python/pyspark/sql/functions.py

```diff
@@ -7596,6 +7721,48 @@ def regexp_extract(str: "ColumnOrName", pattern: str, idx: int) -> Column:
     return _invoke_function("regexp_extract", _to_java_column(str), pattern, idx)
 
 
+@try_remote_functions
+def regexp_extract_all(
+    str: "ColumnOrName", regexp: "ColumnOrName", idx: Optional[Union[int, Column]] = None
```

Comment:
does it make sense to make `idx` also support column name?

Reply:
I don't quite confirm but according to @HyukjinKwon 's suggestion.